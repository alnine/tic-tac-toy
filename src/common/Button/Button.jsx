import React from 'react';
import PropTypes from 'prop-types';
import DefaultButton from './Button.style';

const Button = ({ label, className, disabled, onClick }) => {
  return (
    <DefaultButton className={className} onClick={onClick} disabled={disabled}>
      {label}
    </DefaultButton>
  );
};

export default Button;

Button.propTypes = {
  label: PropTypes.string,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  label: null,
  className: null,
  disabled: false,
  onClick: null,
};
