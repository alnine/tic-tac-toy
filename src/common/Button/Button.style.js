import styled from 'styled-components';

const DefaultButton = styled.button`
  display: inline-block;
  margin: 0;
  padding: 0;
  font: inherit;
  border: none;
  background: none;
  cursor: pointer;
  outline: none;
`;

export default DefaultButton;
