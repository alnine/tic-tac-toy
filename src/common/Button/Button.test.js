import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from './Button';

it('render correctly button component', () => {
  const ButtonComponent = renderer.create(<Button />).toJSON();
  expect(ButtonComponent).toMatchSnapshot();
});

it('render button correctly with null label', () => {
  const ButtonComponent = mount(<Button label={null} />);
  expect(ButtonComponent.prop('label')).toEqual(null);
});

it('check text of button', () => {
  const ButtonComponent = mount(<Button label="Button text" />);
  expect(ButtonComponent.text()).toEqual('Button text');
});

it('check disabled state of button', () => {
  const ButtonComponent = mount(<Button disabled={Boolean(true)} />);
  expect(ButtonComponent.is('[disabled=true]')).toEqual(true);
});

it('check button on availability of class', () => {
  const ButtonComponent = mount(<Button className="test-class" />);
  expect(ButtonComponent.hasClass('test-class')).toEqual(true);
});

it('check the onClick is working', () => {
  const onClick = jest.fn();
  const ButtonComponent = mount(<Button onClick={onClick} />);
  ButtonComponent.simulate('click');
  expect(onClick).toBeCalledTimes(1);
});
