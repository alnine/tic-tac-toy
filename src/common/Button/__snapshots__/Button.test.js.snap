// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly button component 1`] = `
<button
  className="sc-AxjAm kYzbGw"
  disabled={false}
  onClick={null}
/>
`;
